### PR TITLE
ts-warp: init 1.5.4

### PR DESCRIPTION
**Pull request by heywoodlh**
## Description of changes

Adds ts-warp

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ts/ts-warp/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    repo = "ts-warp";
+    rev = finalAttrs.version;
+    hash = "sha256-iozjUHdiiP9Wg86W/LGUucd3sLC8gHMcYAbyAHT7WrQ=";
+  };
+
+  nativeBuildInputs = [ which ];
+
+  PREFIX = "$(out)";
+
+  installFlags = [ "PREFIX=$(out)" ];
```
Comment (Sigmanificient) [inline pkgs/by-name/ts/ts-warp/package.nix:23]: Cant this be passed only once as `makeFlagsArray`?
```suggestion
  makeFlagsArray = [
    "PREFIX = ${placeholder "out"}"
  ];
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ts/ts-warp/package.nix (hunk 1 above)
Comment (heywoodlh) [reply pkgs/by-name/ts/ts-warp/package.nix]: For some reason, the configure phase always tries checking for the default PREFIX (which is `/usr/local /opt $HOME`) which doesn't exist on NixOS and totally ignores the `makeFlagsArray` `$PREFIX`. So the check phase fails on NixOS if I don't set the `$PREFIX` environment variable the way I did. Is there a better way to set an environment variable?

Comment (Sigmanificient) [reply pkgs/by-name/ts/ts-warp/package.nix]: I believe `env.PREFIX = "$(out)";` is the recommended way then

**Review by FliegendeWurst — APPROVED** [review]
There is a superfluous `var` directory in the output. LGTM otherwise

**Pull request merged**